src/app/diary/diary.page.ts

```diff
@@ -0,0 +1,47 @@
+import { Component, OnInit } from '@angular/core'
+import { Router } from '@angular/router'
+import { ToastController } from '@ionic/angular'
+import { DiaryEntry } from '../model/diary-entry'
+import { DiaryService } from '../shared-services/diary/diary.service'
+
+@Component({
+  selector: 'app-diary',
+  templateUrl: './diary.page.html',
+  styleUrls: ['./diary.page.scss'],
+})
+export class DiaryPage implements OnInit {
+  diary: DiaryEntry[]
+
+  constructor(
+    private router: Router,
+    private diaryService: DiaryService,
+    private toastController: ToastController
+  ) {}
+
+  ngOnInit() {}
+
+  async ionViewWillEnter() {
+    this.diary = await this.diaryService.getDiary()
+  }
+
+  async exportDiary() {
+    try {
+      await this.diaryService.exportDiary()
```

Comment:
What I'm missing here is the ability to share the exported diary directly via mail where the recipient and subject are pre-filled with the diary file already added as an attachment – if that is even possible?

Reply:
Might be doable using [this.](https://github.com/katzer/cordova-plugin-email-composer) Haven't looked throughly though.

Reply:
The [Share plugin](https://capacitorjs.com/docs/v2/apis/share) handles that properly. It requires one additional click but you can share it with any other app as well. 

Regarding pre-filled recipient I thought we could add a modal where one can copy the recipient email address. Unfortunately, that's not possible with the share plugin.

| Share dialog (Android 11) | Email (Gmail) |
|---|---|
| ![Screenshot_20220106-104641](https://user-images.githubusercontent.com/12086004/148364409-138eecb8-47c6-491f-bc62-a188a4ceb31e.png) | ![Screenshot_20220106-104653](https://user-images.githubusercontent.com/12086004/148364430-8fea96ea-480c-476d-bee3-48546accd31b.png) |

Reply:
Alright. I'd propose to leave it like that for now then. If really needed, we can always add a mechanism for sending a predefined mail later on.